### PR TITLE
Refactor docker images in EVerest

### PR DESCRIPTION
**Pull request by andistorm**
## Describe your changes

* Add docker image `ghcr.io/everest/everest-core/build-kit-everest-core`
* Refactor workflow `build_and_test.yaml`
  * Deploy docker image `ghcr.io/everest/everest-core/build-kit-everest-core`
  * Use deployed image as build kit image

## Issue ticket number and link

* https://github.com/EVerest/EVerest/issues/178

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

## Requires

* https://github.com/EVerest/everest-ci/pull/22
* https://github.com/EVerest/everest-ci/pull/23

## TODO before merging

- [x] Update everest-ci version to `v1.3.1`

### CODE REVIEW

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -15,7 +15,8 @@ on:
     - cron: '37 13,1 * * *'
 
 env:
-  BUILD_KIT_IMAGE: ghcr.io/everest/build-kit-alpine:v1.2.0
+  DOCKER_REGISTRY: ghcr.io
+  EVEREST_CI_VERSION: refactor-docker-images # LTODO: Update this
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:19]:
```suggestion
  EVEREST_CI_VERSION: v1.3.0
```

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -27,14 +28,49 @@ jobs:
         with:
           path: source
       - name: Run clang-format
-        uses: everest/everest-ci/github-actions/run-clang-format@v1.1.0
+        uses: everest/everest-ci/github-actions/run-clang-format@refactor-docker-images # LTODO: Update this
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:31]:
```suggestion
        uses: everest/everest-ci/github-actions/run-clang-format@v1.3.0
```

File: .github/workflows/build_and_test.yaml [hunk 3]
```diff
@@ -27,14 +28,49 @@ jobs:
...
+    outputs:
+      docker_registry: ${{ env.DOCKER_REGISTRY }}
+      everest_ci_version: ${{ env.EVEREST_CI_VERSION }}
+    steps:
+      - id: check
+        run: |
+          echo "Setting up environment"
+  build-and-push-build-kit:
+    name: Build and Push Build Kit
+    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@refactor-docker-images # LTODO: Update this
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:50]:
```suggestion
    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@v1.3.0
```

**Review by andistorm — COMMENTED** [review]
TODO before merge

File: .github/workflows/build_and_test.yaml [hunk 4]
```diff
@@ -27,14 +28,49 @@ jobs:
...
+    outputs:
+      docker_registry: ${{ env.DOCKER_REGISTRY }}
+      everest_ci_version: ${{ env.EVEREST_CI_VERSION }}
+    steps:
+      - id: check
+        run: |
+          echo "Setting up environment"
+  build-and-push-build-kit:
+    name: Build and Push Build Kit
+    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@v1.3.0
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:50]:
```suggestion
    uses: everest/everest-ci/.github/workflows/deploy-single-docker-image.yml@v1.3.1
```

File: .github/workflows/build_and_test.yaml [hunk 5]
```diff
@@ -27,14 +28,49 @@ jobs:
         with:
           path: source
       - name: Run clang-format
-        uses: everest/everest-ci/github-actions/run-clang-format@v1.1.0
+        uses: everest/everest-ci/github-actions/run-clang-format@v1.3.0
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:31]:
```suggestion
        uses: everest/everest-ci/github-actions/run-clang-format@v1.3.1
```

File: .github/workflows/build_and_test.yaml [hunk 6]
```diff
@@ -15,7 +15,8 @@ on:
     - cron: '37 13,1 * * *'
 
 env:
-  BUILD_KIT_IMAGE: ghcr.io/everest/build-kit-alpine:v1.2.0
+  DOCKER_REGISTRY: ghcr.io
+  EVEREST_CI_VERSION: v1.3.0
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:19]:
```suggestion
  EVEREST_CI_VERSION: v1.3.1
```

**Review by andistorm — COMMENTED** [review]
Update `everest-ci` version again

**Review by hikinggrass — APPROVED** [review]

**Review by Pietfried — APPROVED** [review]

**Pull request merged**